### ISSUE TITLE
Warning: React version not specified in eslint-plugin-react settings (Part 2)

### ISSUE THREAD

**Issue by juliehchung** [jsx-eslint/eslint-plugin-react#3438]
Related to #1955 

I seem to still be having this issue on v7.31.8, discussed in the above issue, and would love some insight if I'm just setting things up incorrectly.
This is a brand new React/TS project being set up using VS Code and Vite, so what you see is all I have currently.
The eslint library is being loaded in successfully, and I don't have any errors, but I am still getting this warning:
![Screen Shot 2022-09-22 at 1 26 35 PM](https://user-images.githubusercontent.com/54191326/191844642-8aaa4ef5-71d0-4d24-896d-7e8bd8187ab4.png)

* I have tried running eslint via the command line, and am still getting the same warning.

The contents of my `.eslintrc.js`:
```
module.exports = {
  root: true,
  env: {
    node: true,
    browser: true
  },
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2022,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    },
    settings: {
      react: {
        version: 'detect'
      },
      'import/resolver': {
        node: {
          paths: ['src'],
          extensions: ['.js', '.jsx', '.ts', '.tsx']
        }
      }
    }
  },
  extends: [
    'eslint:recommended',
    'plugin:react/recommended',
    'plugin:import/recommended',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/recommended',
    'eslint-config-prettier'
  ],
  rules: {
    'react/react-in-jsx-scope': 'off',
    semi: [2, 'always']
  }
};
```

Here is a repro repo where I'm currently having the issue:
https://github.com/juliehchung/test-dash

**Comment by MidKnightXI** [comment]
Can you try this maybe

```json
module.exports = {
  root: true,
  env: {
    node: true,
    browser: true
  },
  settings: {
    react: {
      version: 'detect'
     },
  }
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2022,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true
    },
      'import/resolver': {
        node: {
          paths: ['src'],
          extensions: ['.js', '.jsx', '.ts', '.tsx']
        }
      }
  },
  extends: [
    'eslint:recommended',
    'plugin:react/recommended',
    'plugin:import/recommended',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/recommended',
    'eslint-config-prettier'
  ],
  rules: {
    'react/react-in-jsx-scope': 'off',
    semi: [2, 'always']
  }
};
```

**Comment by MidKnightXI** [comment]
Can you put 

```json
    settings: {
      react: {
        version: 'detect'
      },
    }
```

Outside of your stuff maybe ?

**Comment by juliehchung** [comment]
> 

What do you mean outside of my stuff?

**Comment by juliehchung** [comment]
> Can you put
> 
> ```json
> settings: {
>   react: {
>     version: 'detect'
>   },
>  }
> ```
> 
> Outside of your stuff maybe ?

What do you mean outside of my stuff?

**Comment by MidKnightXI** [comment]
> > Outside of your stuff maybe ?
> 
> What do you mean outside of my stuff?

Sorry if I'm not clear, just edited the message

**Comment by juliehchung** [comment]
> Sorry if I'm not clear, just edited the message

Dang, that was really all it was.. I must have been looking at some outdated references. Appreciate this a lot!

**Comment by ljharb** [comment]
Glad you figured it out!

**Comment by javiercastaneda303** [comment]
insert this code at at .eslintrc.json
{
...
  settings: {
    react: {
      version: 'detect'
    },
   }
...
}

**Comment by MidKnightXI** [comment]
> insert this code at at .eslintrc.json { ... settings: { react: { version: 'detect' }, } ... }

It would actually be like that for json format:

```json
"settings": {
  "react": {
    "version": "detect"
  },
}
```

**Comment by pkpardeep482** [comment]
Its not working after putting 

"settings": {
  "react": {
    "version": "detect"
  },
}

**Comment by ljharb** [comment]
@pkpardeep482 if you upgrade to the latest version and it's still not working, please file a new issue.
